Backend/app/rider-platform/rider-app/Main/src/Domain/Action/Dashboard/FRFSTicket.hs

```diff
@@ -16,38 +16,281 @@ where
 import qualified API.Types.RiderPlatform.Management.FRFSTicket
 import qualified BecknV2.FRFS.Enums
 import Dashboard.RiderPlatform.FRFSTicket
+import qualified Data.ByteString as BS
+import qualified Data.ByteString.Lazy as LBS
+import Data.Csv
+import qualified Data.List.NonEmpty as NE
 import Data.OpenApi (ToSchema)
 import qualified Data.Text
+import qualified Data.Vector as V
 import qualified Domain.Types.Merchant
+import Domain.Types.Route
+import Domain.Types.Station
 import qualified Environment
+import qualified EulerHS.Language as L
 import EulerHS.Prelude hiding (id)
+import Kernel.External.Maps.Types
 import qualified Kernel.Prelude
+import Kernel.Types.APISuccess (APISuccess (..))
 import qualified Kernel.Types.APISuccess
 import qualified Kernel.Types.Beckn.Context
+import Kernel.Types.Common
+import Kernel.Types.Error
+import Kernel.Types.GuidLike (generateGUID)
 import qualified Kernel.Types.Id
-import Servant
+import Kernel.Types.Time (getCurrentTime)
+import Kernel.Types.TimeBound
+import Kernel.Utils.Common (fromMaybeM, throwError)
+import Kernel.Utils.Logging (logInfo)
+import Servant hiding (Header, throwError)
+import qualified Storage.CachedQueries.Merchant as QM
+import qualified Storage.CachedQueries.Merchant.MerchantOperatingCity as CQMOC
+import Storage.Queries.Route as QRoute
+import Storage.Queries.RouteExtra as RE
+import Storage.Queries.RouteStopFare as QRSF
+import Storage.Queries.RouteStopMapping as QRSM
+import Storage.Queries.Station as QStation
+import Storage.Queries.StationExtra as SE
 import Tools.Auth
 
-getFRFSTicketFrfsRoutes :: (Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant -> Kernel.Types.Beckn.Context.City -> Kernel.Prelude.Int -> Kernel.Prelude.Int -> BecknV2.FRFS.Enums.VehicleCategory -> Environment.Flow [API.Types.RiderPlatform.Management.FRFSTicket.FRFSRouteAPI])
-getFRFSTicketFrfsRoutes _merchantShortId _opCity limit offset vehicleType = do error "Logic yet to be decided" limit offset vehicleType
+getFRFSTicketFrfsRoutes :: (Kernel.Types.Id.ShortId Domain.Types.Merchant.Merchant -> Kernel.Types.Beckn.Context.City -> Kernel.Prelude.Maybe Data.Text.Text -> Kernel.Prelude.Int -> Kernel.Prelude.Int -> BecknV2.FRFS.Enums.VehicleCategory -> Environment.Flow [API.Types.RiderPlatform.Management.FRFSTicket.FRFSRouteAPI])
+getFRFSTicketFrfsRoutes merchantShortId opCity searchStr limit offset vehicleType = do
+  let limitVal = toInteger $ fromMaybe 10 (Just limit)
```

Comment:
Why fromMaybe is required ? It's already a mandatory field on UI.
Yo may remove fromMaybe.